app/javascript/lib/cell-faceting.js

```diff
@@ -182,6 +168,110 @@ function mergeFacetsResponses(newRawFacets, prevCellFaceting) {
   return mergedRawFacets
 }
 
+/** Omit any filters that match 0 cells in the current clustering */
+function trimNullFilters(cellFaceting) {
```

Comment:
On first pass, I think I agree with this approach (removing empty entries) but I wonder if there's a scientific value for seeing that a combination of filters results in populations being reduced to 0?

Reply:
Yes, I think there's definitely scientific value in seeing when a selection yields a 0-count for a given filter.  This "null filter" trimming addresses something a bit different.  

It turns out that _some filters have a 0-count without any selection having been made_.  For example, if you go to [SCP1671 on production](https://singlecell.broadinstitute.org/single_cell/study/SCP1671/cellular-and-transcriptional-diversity-over-the-course-of-human-lactation), there's a filter shown for "animal cell".  Selecting and deselecting the "animal cell" filter does nothing, unlike all the other filters in that facet.  That's because while "animal cell" is an annotation group value in the metadata file, no cell in that clustering has the group value "animal cell".

That's a pre-existing issue, but it becomes severe once filter counts are shown.  Such "null groups" would also affect the scatter plot legend and DE panel, but get omitted by code that builds `countsByLabel`.  This measure in `trimNullFilters` applies a similar approach to omitting those null groups in filters.

Reply:
Ah - thank you for that explanation.  That makes total sense!